### DISCUSSION TITLE
Decawave module firmware

### DISCUSSION THREAD

**Discussion by viorels** [bitcraze/discussions#72] [Q&A]
I am looking for indoor localization solutions and I initially found Decawave DWM1001dev modules (and that lead me to Bitcraze eventually). The default firmware on those modules is OK, but the filtering and limitation of max 4 anchors used at once for localization is sometimes not good enough.
I'm curious how do you guys use the modules as it seems your solution is more precise.
Did you create an alternative firmware for Decawave module itself? Is it open source? Thanks

**Comment by krichardsson** [comment]
Hi!

Yes, we have written all firmware. There are two main modes Two Way Ranging and Time Difference of Arrival. In TWR, the ranging sequence is initiated by the drone, targeting one anchor at a time. This mode is limited to one drone and 8 anchors. For TDoA the drone is passively listening while the anchors are transmitting (similar to GPS). In this mode there is no limit to the number of drones and (in theory) 255 anchors, but the system has mainly been tested with 8 anchors.

We do not have an alternative firmware for the Deckawave module, only for our own products, but it is open source if you are interested in porting it. I do think it would require quite some work, but maybe you can get some inspiration from our work?

Main repositories are:
Anchor fw: https://github.com/bitcraze/lps-node-firmware
Driver for the DW1000: https://github.com/bitcraze/libdw1000
Drone fw: https://github.com/bitcraze/crazyflie-firmware

You can also read about the system in general here https://www.bitcraze.io/documentation/system/ 
implementation here https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/ and here https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/loco-positioning-system/
